src/Chainweb/Pact5/TransactionExec.hs

```diff
@@ -303,7 +302,7 @@ applyLocal logger maybeGasLogger coreDb txCtx spvSupport cmd = do
     -- anyone's workflow
     , FlagAllowReadInLocal
     , FlagRequireKeysetNs
-    ]
+    ] `Set.union` (if guardCtx chainweb228Pact txCtx then Set.empty else Set.singleton FlagDisablePact51)
```

Comment:
Please make a reusable function for this that can be called by `applyCmd` and `applyLocal`, similar to how it's done in `Pact4.TransactionExec`, so that we don't miss adding this to both places in future.